src/instana/instrumentation/pyramid.py

```diff
@@ -78,16 +78,18 @@ def __call__(self, request: "Request") -> "Response":
                         span.set_attribute(
                             SpanAttributes.HTTP_STATUS_CODE, response.status_int
                         )
-
-                        if 500 <= response.status_int:
-                            if response.exception:
-                                span.record_exception(response.exception)
-                            span.assure_errored()
-
+                        if response.status_code >= 500:
+                            handle_exception(span, response)
                 return response
 
     INSTANA_TWEEN = __name__ + ".InstanaTweenFactory"
 
+    def handle_exception(span, response: Union[Response, HTTPException]) -> None:
```

Comment:
Moving the `Response` import back to inside the `if TYPE_CHECKING` block, you should use double quotes in the type annotation: 

```suggestion
    def handle_exception(span, response: Union["Response", HTTPException]) -> None:
```